app/backplane/deployment_module/include/c_pyro_control_observer.h

```diff
@@ -0,0 +1,64 @@
+#ifndef C_PYRO_CONTROL_OBSERVER_H
+#define C_PYRO_CONTROL_OBSERVER_H
+
+#include <array>
+#include <f_core/device/c_gpio.h>
+#include <f_core/os/c_file.h>
+#include <f_core/utils/c_observer.h>
+
+
+class CPyroControlObserver : public CObserver {
+public:
+    CPyroControlObserver() = default;
+
+    ~CPyroControlObserver() override = default;
+
+    /**
+     * See parent docs
+     * @param ctx Alert number
+     */
+    void Notify(void* ctx) override;
+
+private:
+    struct PyroPair {
+        CGpio sense;
+        CGpio ctrl;
+    };
+
+
+    std::array<CGpio, 4> de9Gpios{
+        CGpio(GPIO_DT_SPEC_GET(DT_ALIAS(gpio0), gpios)),
+        CGpio(GPIO_DT_SPEC_GET(DT_ALIAS(gpio1), gpios)),
+        CGpio(GPIO_DT_SPEC_GET(DT_ALIAS(gpio2), gpios)),
+        CGpio(GPIO_DT_SPEC_GET(DT_ALIAS(gpio3), gpios))
+    };
+
+    CFile logFile{"deployment_log.txt", CFile::READ_WRITE_FLAG | CFile::CREATE_FLAG | CFile::APPEND_FLAG};
```

Comment:
Any reason for not using CFlightLog, would get you timestamps for free so you don't have to add them manually like you do in this .cpp. 

Im seeing that you use an rtc time which it doesnt currently support but it has overloads for custom integerr timestamps, it wouldnt be a big jump to add an overload for custom string timestamp. imo that file is for exactly this purpose of needing to write timestamped log messages. CFile does the trick but it feels like its no necessarily the best tool for the job

Reply:
Also, we should probably make it so this filename is modified so it doesn't overwrite itself

Reply:
Also also, does that filename work? usually i have to do something like `/lfs/deployment_log.txt` or else zephyr doesn't know what mount point to put it at

Reply:
Pretty sure we just append the log. Also, yea forgot the mp. Using flight log now

Reply:
word